### ISSUE TITLE
Add `severity.parseError` option to control parse error severity

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#2112]
Currently, `markuplint` treats all parse errors as critical errors, causing the build process to fail. However, in some cases, it is desirable to treat parse errors with different severity levels, such as warnings or ignoring them altogether. This feature request proposes the addition of a `severity.parseError` option, which allows users to define the severity of parse errors.

## Proposal:
1. **CLI Option**:  
   Add a new CLI option `--parse-error-severity` to control the severity of parse errors. It should accept the following values:
   - `error` (default): Treat parse errors as critical and fail the build.
   - `warning`: Treat parse errors as warnings, but do not fail the build.
   - `off`: Ignore parse errors completely.

   Example usage:
   ```sh
   markuplint path/to/*.html --parse-error-severity warning
   ```

2. **Configuration Option**:  
   Extend the configuration file to include a `severity.parseError` option. This option will control the severity of parse errors at the configuration level.

   Example configuration:
   ```json
   {
     "severity": {
       "parseError": "warning"
     }
   }
   ```

3. **Error Handling Update**:  
   Modify the internal logic to respect the configured severity level when a parse error occurs. If the severity is set to `warning`, the error should be logged as a warning but should not fail the process. If set to `off`, parse errors should be ignored.

## Benefits:
- **Flexibility**: Users can choose how strict they want the linter to be regarding parse errors.
- **Better Development Experience**: Developers can opt to treat parse errors as warnings in non-critical environments, improving workflow during development without disrupting the build process.

## Expected Outcome:
After implementing this feature, users will have more control over how parse errors are handled, improving the tool's flexibility and usability in different development environments.
